src/request-pipeline/index.js

```diff
@@ -113,7 +113,11 @@ function createReqOpts (ctx) {
 function sendResponseHeaders (ctx) {
     var headers = headerTransforms.forResponse(ctx);
 
-    ctx.res.writeHead(ctx.destRes.statusCode, headers);
+    if (ctx.contentInfo.requireProcessing && ctx.destRes.statusCode === 204 && headers['content-length'] > 0)
```

Comment:
If we not process page with 204 status, then hammerhead will be not initialized. [Description](https://github.com/DevExpress/testcafe-hammerhead/issues/306)